azext_edge/edge/providers/orchestration/resources/schema_registries.py

```diff
@@ -159,3 +162,169 @@ def delete(self, name: str, resource_group_name: str, confirm_yes: Optional[bool
+            schema_format = SchemaFormat[schema_format].full_value
+            resource = {
+                "properties": {
+                    "format": schema_format,
+                    "schemaType": schema_type,
+                    "description": description,
+                    "displayName": display_name,
+                },
+            }
+            schema = self.ops.create_or_replace(
+                resource_group_name=resource_group_name,
+                schema_registry_name=schema_registry_name,
+                schema_name=name,
+                resource=resource
+            )
+            logger.info(f"Created schema {name}.")
+            # TODO: maybe add in an exception catch for auth errors
+            self.add_version(
+                name=schema_version,
+                schema_content=schema_content,
+                schema_name=name,
+                schema_registry_name=schema_registry_name,
+                resource_group_name=resource_group_name,
+                description=schema_version_description,
+                current_console=c
+            )
+            logger.info(f"Added version {schema_version} to schema {name}.")
+            return schema
+
+    def show(self, name: str, schema_registry_name: str, resource_group_name: str) -> dict:
+        return self.ops.get(
+            resource_group_name=resource_group_name,
+            schema_registry_name=schema_registry_name,
+            schema_name=name
+        )
+
+    def list(self, schema_registry_name: str, resource_group_name: str) -> Iterable[dict]:
+        return self.ops.list_by_schema_registry(
+            resource_group_name=resource_group_name, schema_registry_name=schema_registry_name
+        )
+
+    def delete(
+        self,
+        name: str,
+        schema_registry_name: str,
+        resource_group_name: str,
+        confirm_yes: Optional[bool] = None,
+    ):
+        if not should_continue_prompt(confirm_yes=confirm_yes):
+            return
+
+        with console.status("Working..."):
+            return self.ops.delete(
+                resource_group_name=resource_group_name,
+                schema_registry_name=schema_registry_name,
+                schema_name=name
+            )
+
+    def add_version(
+        self,
+        name: int,
+        schema_name: str,
+        schema_registry_name: str,
+        resource_group_name: str,
+        schema_content: str,
+        description: Optional[str] = None,
+        current_console: Optional[Console] = None,
+    ) -> dict:
+        from ....util import read_file_content
+
+        if name < 0:
+            raise InvalidArgumentValueError("Version must be a positive number")
+
+        try:
+            logger.debug("Processing schema content.")
+            schema_content = read_file_content(schema_content)
+        except FileOperationError:
+            logger.debug("Given schema content is not a file.")
+            pass
```

Comment:
Should this pass or throw an error?

Reply:
pass cause the flow is:
1. try to open the schema content (assuming file)
2. fail -> use the schema content as the inline

now - can I somehow do something  bad with this (1)